### PR TITLE
qt6: use strawberryperl on mingw and clang-cl

### PR DESCRIPTION
**Pull request by ericLemanissier**
also, fix linter

fixes conan-io/conan-center-index#9969

Specify library name and version:  **qt/6.***

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def build_requirements(self):
         self.build_requires("cmake/3.22.0")
         self.build_requires("ninja/1.10.2")
         self.build_requires("pkgconf/1.7.4")
-        if self._is_msvc:
+        if self.settings.os == "Windows":
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:351]: this is the only part of PR which is an actual change

**Comment by AndreyMlashkin** [comment]
@ericLemanissier I am getting another error now. However, I think this PR should be merged anyway, as it is definitelly an imrovement

**Comment by ericLemanissier** [comment]
please give me the full log. the more fixes we put in a single PR, the better

**Comment by AndreyMlashkin** [comment]
Okey, here we go
[qt_build_log.txt](https://github.com/conan-io/conan-center-index/files/8363125/qt_build_log.txt)

**Comment by ericLemanissier** [comment]
qt recipes assumes that clang on windows is not clang-cl: https://github.com/conan-io/conan-center-index/pull/9971/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R480
If you want to build qt using clang-cl, then you need to make this function return `win32-clang-msvc`. I would have done it with `if self.settings.compiler == "Visual Studio" and self.settings.compiler.toolset="ClangCL"`, but as far as I undstood your previous issues, it does not cover all use cases. Do we have a reliable way to discriminate between clang and clang-cl ? Are we waiting for a conan feature ? a new compiler setting ?

**Comment by AndreyMlashkin** [comment]
@SSE4, correct me, if I am wrong, the correct way for checking is:
self.settings.compiler == "clang" and self.settings.os == "Windows"

**Comment by ericLemanissier** [comment]
no, clang and windows can be either clang-cl or regular clang

**Review by AndreyMlashkin — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@intelligide @jgsogo @SpaceIm 
please approve

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**